src/amo/components/HomepageShelves/index.js

```diff
@@ -0,0 +1,127 @@
+/* @flow */
+import * as React from 'react';
+import config from 'config';
+
+import LandingAddonsCard from 'amo/components/LandingAddonsCard';
+import LoadingText from 'amo/components/LoadingText';
+import {
+  INSTALL_SOURCE_FEATURED,
+  INSTALL_SOURCE_FEATURED_COLLECTION,
+  LANDING_PAGE_EXTENSION_COUNT,
+  LANDING_PAGE_THEME_COUNT,
+} from 'amo/constants';
+import translate from 'amo/i18n/translate';
+import type { ResultShelfType } from 'amo/reducers/home';
+import type { I18nType } from 'amo/types/i18n';
+
+import './styles.scss';
+
+type DefaultProps = {|
+  placeholderCount: number,
+|};
+
+type InternalProps = {|
+  i18n: I18nType,
+|};
+
+type Props = {|
+  ...DefaultProps,
+  ...InternalProps,
+  loading: boolean,
+  placeholderCount?: number,
+  shelves: Array<ResultShelfType>,
+|};
+
+class HomepageShelves extends React.Component<Props> {
+  /* eslint-disable react/no-array-index-key */
```

Comment:
We shouldn't need this. I can see it's needed for the key for the loading state shelves, but you can do that another way. I'll add a suggestion to that.